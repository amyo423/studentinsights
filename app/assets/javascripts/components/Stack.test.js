import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stack from './Stack';


export function testProps(props) {
  return {
    stacks: [
      {count: 18, color: 'red'},
      {count: 8, color: 'orange'}
    ],
    scaleFn(count) { return count / (18+8); },
    labelFn(count) { return count; },
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(<Stack {...props} />, el);
});

it('snapshots', () => {
  const props = testProps();
  const tree = renderer
    .create(<Stack {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});