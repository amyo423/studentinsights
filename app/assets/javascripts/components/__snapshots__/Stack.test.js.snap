// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  style={
    Object {
      "color": "black",
      "cursor": "default",
      "display": "inline-block",
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "red",
          "height": "100%",
          "left": "0%",
          "position": "absolute",
          "width": "69%",
        }
      }
    />
    <div
      style={
        Object {
          "height": "100%",
          "left": "0%",
          "position": "absolute",
          "width": "69%",
        }
      }
    >
      18
    </div>
  </div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "orange",
          "height": "100%",
          "left": "69%",
          "position": "absolute",
          "width": "31%",
        }
      }
    />
    <div
      style={
        Object {
          "height": "100%",
          "left": "69%",
          "position": "absolute",
          "width": "31%",
        }
      }
    >
      8
    </div>
  </div>
</div>
`;
